### DISCUSSION TITLE
Consider removing all `repo` interfaces

### DISCUSSION THREAD

**Discussion by nazrinharris** [nazrinharris/pecunia#51] [Discussion]
After asking for code review, it seems like interfaces aren't really needed. Implementations can be directly mocked with `mocktail`. I'll keep all the datasource interfaces though. I also would need to think about if my "layering" makes sense. 

As of now, +1 for not making sense, take a look at `createTransaction`, basically we're going through a bunch of abstraction layers, which don't seem to do much, other than passing back the `TaskEither`.

Basically +1 for in favor of removing repo and vice versa

**Comment by nazrinharris** [comment]
-1 because interfaces are useful for legibility. I found it extremely convenient to see the all the methods and its arguments at the top. It's like headers of some sort. The addition of interfaces isn't that inconvenient to begin with.

**Reply by nazrinharris** [reply]
Legibility? Shnawg use comments for that stuff. Maybe even a VSCode extension to generate the stuff

**Comment by nazrinharris** [comment]
+1 Kinda to refactor the entire project's architecture to something much simpler, but highly dependent on `riverpod`. Something like [this project by "krunchy" on discord](https://github.com/runionke/indegoal/blob/main/lib/providers/auth_provider.dart). (high dependence on `riverpod` isn't necessarily a bad thing). 

I imagine I can still somehow incorporate the current folder structure, but combine the three layers (`repo`, `data`, `dao_tables`) into maybe two layers (`providers`, `dao_tables`).

The end goal of the architecture I'm pursuing is one which is easily navigable. Some friction is fine, but anyone looking at it (specifically future me) should be able to intuitively understand where something is supposed to go.

**Comment by nazrinharris** [comment]
+1 Here's a structure I'm envisioning:

1. `data` layer. Includes the `transactions_dao_tables.dart` in a `local_ds` and a `remote_ds` if necessary.
2. `provider` layer. Communicates with the `data` layer. Most likely would be one provider per method of sorts. 

General intuition is still available, lowest layer is `data`, then `provider`, then `presentation`. It also simplifies the structure a lot.
